### PR TITLE
Add some documents for developers

### PR DESCRIPTION
**Pull request by Kaiser-Yang**
We add documents for developers, so related part in `README.md` will be removed. Besides, rename the option in `cmake`: `mca_BUILD_SHARE_LIB` -> `mca_BUILD_SHARED_LIB`.

See #120.

### CODE REVIEW

File: doc/en/developer/codeWrite.md [hunk 1]
```diff
@@ -0,0 +1,17 @@
+# How to write my own code
+Once you clone the repository, you should make sure that you are on the `develop` branch, you can
+use `git checkout develop` to convert to `develop` branch. After this, you'd better run `git pull`
+to make sure the `develop` branch is up to date.
+
+Usually, you cannot commit on `develop` and `master` branches (You can commit locally, but you
+cannot push, and you can only make updates on these branches through `pull request`). So if you
+want your codes be submitted to the repository, you can create a new branch depending on `develop`
```
Comment (Kaiser-Yang) [inline doc/en/developer/codeWrite.md:8]: update with: want your code to be submitted...

File: doc/en/developer/codeWrite.md [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+(through `git checkout -b new-branch-name develop`) and write your own codes on the new branch. Of
```
Comment (Kaiser-Yang) [inline doc/en/developer/codeWrite.md:9]: codes --> code

File: doc/en/developer/codeWrite.md [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+course, if you are not in the developer of the repository, you need folk this repository, and update
```
Comment (Kaiser-Yang) [inline doc/en/developer/codeWrite.md:10]: update with: if you are not one of the developers of the repository...

File: doc/en/developer/codeWrite.md [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+on your own repository, then create a pull request depending on the folk repository.
```
Comment (Kaiser-Yang) [inline doc/en/developer/codeWrite.md:11]: folk --> folked

File: doc/en/developer/projectBuild.md [hunk 5]
```diff
@@ -0,0 +1,63 @@
+# How to build the project
+Once you finish your codes on your new branch, you may need to compile the codes to make sure the
```
Comment (Kaiser-Yang) [inline doc/en/developer/projectBuild.md:2]: codes --> code

**Review by luckygalaxy666 — APPROVED** [review]

File: doc/en/developer/projectBuild.md [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+# All the options for cmake
+|                                      |   |
+| -                                    | - |
+| <nobr>`mca_BUILD_SHARED_LIB`</nobr>  | Whether or not to build shared libraries. Default value `on` |
+| <nobr>`mca_BUILD_TEST`</nobr>        | Whether or not to build unit test. Default value `off` |
+| <nobr>`mca_GENERATE_COVERATE`</nobr> | Whether or not to generate code coverage report. Default value `off` |
+| <nobr>`mca_BUILD_EXAMPLE`</nobr>     | Whether or not to build examples. Default value `on` |
+
+# Write and run the unit test
+Although you compile your codes successfully, you should be aware of that the codes which is
```
Comment (Kaiser-Yang) [inline doc/en/developer/projectBuild.md:30]: codes --> code

File: doc/en/developer/projectBuild.md [hunk 7]
```diff
@@ -0,0 +1,63 @@
...
+compiled successfully may not work properly. So you need write unit tests for the every function you
+write. So you need create a file which is related with your new codes. The filename usually is the
+origin base filename followed by `_test.cpp`. If there have been already one, you just need append
+your unit tests codes. If you finish writing unit tests, you can re-build the project through step
```
Comment (Kaiser-Yang) [inline doc/en/developer/projectBuild.md:34]: codes --> code

File: doc/en/developer/projectBuild.md [hunk 8]
```diff
@@ -0,0 +1,63 @@
...
+`1-3` above. And change you directory to `build/test` (after building successfully), then run
+`ctest --output-on-failure`, this will start run all the test units. If all the tests pass, your
+code may be OK. If there are some tests failed, you need check the codes you write (both test code
+and the source code), and update them to pass all the tests.
+
+# Write examples and documents for public API
+All the public (used by users) functions or methods you add for the project should be documented.
+You should update the document `doc/en/user/index.md` to add a new index, and then add a new file
+related to the new index. If there already is a file for file containing the new API, you just need
+update the existed file. All the examples will be built and compiled to an executable in the
```
Comment (Kaiser-Yang) [inline doc/en/developer/projectBuild.md:44]: All the examples --> Every example

File: doc/en/developer/projectBuild.md [hunk 9]
```diff
@@ -0,0 +1,63 @@
...
+# How to generate code coverage report locally
+You may want to see the code coverage locally. You can do those below to generate the `html` files
+for code coverage report after building and compiling (make sure `mca_BUILD_TEST` and
+`mca_GENERATE_COVERATE` are `on`):
+
+1. Change your directory to `build/test`.
+2. Run `./mca_unit_test`.
+3. Change your directory to `build`.
+4. Run `lcov --capture --directory . --output-file coverage.info` to generate report.
+5. Run `genhtml --output-directory coverage_report coverage.info` to generate html files.
```
Comment (Kaiser-Yang) [inline doc/en/developer/projectBuild.md:56]: html --> `html`

File: doc/en/developer/codeSubmit.md [hunk 10]
```diff
@@ -0,0 +1,34 @@
+# How to submit my codes to the repository
+Once you finish your code, and pass all the tests. You needn't worry about code formatting, git
```
Comment (Kaiser-Yang) [inline doc/en/developer/codeSubmit.md:2]: git action --> `git action`

File: doc/en/developer/codeSubmit.md [hunk 11]
```diff
@@ -0,0 +1,34 @@
...
+action will automatically format the code for you when you create a request to develop branch or
```
Comment (Kaiser-Yang) [inline doc/en/developer/codeSubmit.md:3]: request --> pull request.

File: doc/en/developer/codeSubmit.md [hunk 12]
```diff
@@ -0,0 +1,34 @@
...
+master branch. What you need to do is to make sure all you changed have been committed locally,
```
Comment (Kaiser-Yang) [inline doc/en/developer/codeSubmit.md:4]: you changed --> your changes

File: doc/en/developer/codeSubmit.md [hunk 13]
```diff
@@ -0,0 +1,34 @@
...
+and make sure the commit messages are readable (meet the rules).
```
Comment (Kaiser-Yang) [inline doc/en/developer/codeSubmit.md:5]: add a link for the rules.

File: doc/en/developer/codeSubmit.md [hunk 14]
```diff
@@ -0,0 +1,34 @@
...
+
+Now you can create a branch remotely, you can use `git push origin -u your-branch-name` to push your
+branch to remote with the same name. After this, you can create a `pull request` to merge into
+`develop` branch ( **choose the right branch**). Write the title and the description. The changes
+will be checked by `git actions`. Once it passed, you just need to wait the collaborators to accept
```
Comment (Kaiser-Yang) [inline doc/en/developer/codeSubmit.md:10]: `git action` --> `git action`s
it --> they

File: doc/en/developer/codeSubmit.md [hunk 15]
```diff
@@ -0,0 +1,34 @@
...
+your changes or give you some new suggestions.
+
+# How to update when I was told
+Sometime, your codes may be add some reviews by the collaborators, you need update your codes to
+resolve these reviews.
+
+You can update your local codes to meet the collaborators' suggestions. After this, you can commit,
```
Comment (Kaiser-Yang) [inline doc/en/developer/codeSubmit.md:17]: codes --> code

File: doc/en/developer/codeSubmit.md [hunk 16]
```diff
@@ -0,0 +1,34 @@
...
+for the review.
+
+**Note: the below is not recommended.**
+
+If you think the new commit can be submitted without creating a new commit, after you update your
+codes, you can use `git add` to add your codes, then `git commit --amend` to let the new changes be
+in the last commit. After this, you must use `git push -f` to update the remote forcedly.
+
+# When to Merge Develop Branch into Master Branch
+This will depend on me. Usually, when some new features finished and tested, I'll create a `release`
```
Comment (Kaiser-Yang) [inline doc/en/developer/codeSubmit.md:29]: finished and tested --> are finished and tested

**Review by Quentin9922 — APPROVED** [review]

**Pull request merged**